generators/server/templates/quarkus/src/main/java/package/web/rest/vm/EnvVM.java.ejs

```diff
@@ -0,0 +1,91 @@
+<%#
+ Copyright 2020-2021 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%=packageName%>.web.rest.vm;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+
+@RegisterForReflection
+public class EnvVM {
+  private final Collection<String> activeProfiles;
+  private final Collection<PropertySource> propertySources;
+
+  public EnvVM(Collection<String> activeProfiles, Collection<PropertySource> propertySources) {
+    this.activeProfiles = activeProfiles;
+    this.propertySources = propertySources;
+  }
+
+  public Collection<String> getActiveProfiles() {
+    return activeProfiles;
+  }
+
+  public Collection<PropertySource> getPropertySources() {
+    return propertySources;
+  }
+
+  public static class PropertySource {
+    private final String name;
+    private final Map<String, PropertyValue> properties;
+
+    public PropertySource(String name, Map<String, String> properties) {
+      this.name = name;
+      this.properties = properties.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, PropertyValue::new));
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public Map<String, PropertyValue> getProperties() {
+      return properties;
+    }
+
+    public static class PropertyValue {
+      private final String value;
+
+      public PropertyValue(Map.Entry<String, String> entry) {
+        this.value = retrieveValue(entry);
+      }
+
+      public PropertyValue(String value) {
+        this.value = value;
+      }
+
+      public String getValue() {
+        return value;
+      }
+
+      private String retrieveValue(Map.Entry<String, String> entry) {
+        final String obfuscatedValue = "******";
+        String[] prefixes = { "password", "credentials", "credential", "secret", "secrets" };
+
+        for (String prefix : prefixes) {
+          if (StringUtils.containsIgnoreCase(entry.getKey(), prefix)) {
```

Comment:
Please use vanilla Java for this kind of operation.
IMHO, there is no need to import a lib (ie. commons-lang) for 1 statement.
`entry.getKey().toLowerCase().contains(prexis)` should do the work.
Otherwise, RegExp would do the job quite easily ;)